dashboard/src/components/AppList/AppList.tsx

```diff
@@ -30,19 +33,38 @@ class AppList extends React.Component<IAppListProps, IAppListState> {
     this.setState({ filter });
   }
 
-  public componentWillReceiveProps(nextProps: IAppListProps) {
+  public componentDidUpdate(prevProps: IAppListProps) {
     const {
       apps: { error, listingAll },
       fetchApps,
       filter,
       namespace,
     } = this.props;
     // refetch if new namespace or error removed due to location change
-    if (nextProps.namespace !== namespace || (error && !nextProps.apps.error)) {
-      fetchApps(nextProps.namespace, listingAll);
+    if (prevProps.namespace !== namespace || (error && !prevProps.apps.error)) {
+      fetchApps(namespace, listingAll);
     }
-    if (nextProps.filter !== filter) {
-      this.setState({ filter: nextProps.filter });
+    if (prevProps.filter !== filter) {
+      this.setState({ filter });
+    }
+    if (prevProps.apps !== this.props.apps && this.props.apps.listOverview) {
+      this.props.apps.listOverview.forEach(app => {
+        if (
+          !this.props.updatesInfo[
+            getChartUpdatesKey(
+              app.chartMetadata.name,
+              app.chartMetadata.version,
+              app.chartMetadata.appVersion,
+            )
+          ]
+        ) {
+          this.props.getChartUpdates(
```

Comment:
these dependent calls are difficult inside a component, I think it would be better to create a new action, such as `fetchAppsWithUpdatesInfo` that is able to dispatch and await on the apps and then dispatch the `getChartUpdates` action. This approach has the benefit of keeping the actions decoupled (`fetchApps` and `getChartUpdates` are still separate, but `fetchAppsWithUpdatesInfo` is able to handle the dependency between them) and avoid the need for this complicated, potentially buggy (as per issues stated in https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html), logic in the component.